font/src/ft/mod.rs

```diff
@@ -63,22 +65,20 @@ struct FallbackList {
     coverage: CharSet,
 }
 
-struct Face {
-    ft_face: freetype::Face,
-    key: FontKey,
+struct FaceLoadingProperties {
     load_flags: freetype::face::LoadFlag,
     render_mode: freetype::RenderMode,
     lcd_filter: c_uint,
     non_scalable: Option<FixedSize>,
     has_color: bool,
     pixelsize_fixup_factor: Option<f64>,
+    ft_face: Rc<FTFace>,
```

Comment:
Okay, before going deeper into this code I have the question what the benefit is of having a reference to the face in the hashmap here, rather than just storing the PathBuf to the font?

Is the goal solely to make it so the reference is dropped, as soon as `FaceLoadingProperties` is dropped, destroying the face?

Are there any concerns about just trashing the `ft_faces` hashmap with a bazillion paths to faces that don't even exist anymore?

Reply:
>Is the goal solely to make it so the reference is dropped, as soon as FaceLoadingProperties is dropped, destroying the face

Yes, that was the idea.

>Is the goal solely to make it so the reference is dropped, as soon as FaceLoadingProperties is dropped, destroying the face

IIRC paths are cheaper to keep around compared to faces, since it’s max 4KB per path, but it is a worst case.

Reply:
Have you ever checked how much is needed to stare a face in Rust? 4KB seems like a lot.

Reply:
Oh, in addition having reference to a `FTFace` simplify the logic, since you need `FTFace` to perform rasterizing, so using indirection with `PathBuf` is slower, since you need to query `HashMap` each time to get a proper face. I have an idea on how to solve your zombie `PathBuf` issue though.

Reply:
Ok I was a bit wrong that I know a way to remove this zombies automatically. However you repro an issue I'm fixing by doing `cat /dev/urandom` and starting changing your font size, you should notice that master uses way more RAM on average.